File: Lista 1/Craps.java
import java.util.Random;

public class Craps {
    public static void main(String[] args){
        // Estagio 1
        int dado1 = jogarDado(), dado2 = jogarDado(), ponto = dado1+dado2;
        System.out.println("ESTAGIO 1\nDado 1: " + dado1 + " | Dado 2: " + dado2);
        System.out.println("Ponto: " + ponto);
        
        if (ponto == 7 || ponto == 11)
            System.out.println("Parabens, voce venceu!");

        else if (ponto == 2 || ponto == 3 || ponto == 12)
            System.out.println("Que pena, voce perdeu!");

        // Estagio 2
        else{
            int cont = 0;
            System.out.println("\nESTAGIO 2");
            while (true){
                dado1 = jogarDado();
                dado2 = jogarDado();
                System.out.println("Dado 1: " + dado1 + " | Dado 2: " + dado2);
                System.out.println("Jogada " + (++cont) + ": " + (dado1+dado2) + "\n");
                
                if (dado1+dado2 == ponto){
                    System.out.println("Parabens, voce venceu!");
                    break;
                }
                
                else if (dado1+dado2 == 7){
                    System.out.println("Que pena, voce perdeu!");
                    break;
                }
            }
        }
    }

    public static int jogarDado(){
        Random rand = new Random();
        return 1 + rand.nextInt(6);
    }
}

File: Lista 2/Craps.py
def jogarDados():
    from random import randint
    return randint(1, 6), randint(1, 6)

def main():
    # Estagio 1
    dados = jogarDados()
    ponto = sum(dados)
    print(f'ESTAGIO 1\nDado 1: {dados[0]} | Dado 2: {dados[1]}\nPonto: {ponto}')

    if ponto in (7, 11):
        print('Parabens, voce venceu!')
        
    elif ponto in (2, 3, 12):
        print('Que pena, voce perdeu!')

    # Estagio 2
    else:
        cont = 1
        print('\nESTAGIO 2')
        while(True):
            dados = jogarDados()
            print(f'Dado 1: {dados[0]} | Dado 2: {dados[1]}\nJogada {cont}: {sum(dados)}\n')

            if sum(dados) == ponto:
                print('Parabens, voce venceu!')
                break

            elif sum(dados) == 7:
                print('Que pena, voce perdeu!')
                break


if __name__ == '__main__':
    main()